### ISSUE TITLE
Sample code that runs perfectly in JS API 4.17 does not work as expected in JS API 4.18

### ISSUE THREAD

**Issue by MiriEshel** [Esri/jsapi-resources#268]
Hello everyone,

One of our customers is using JS API 4.18 in Angular environment.

He claims that running the code in: https://developers.arcgis.com/javascript/latest/sample-code/sandbox/index.html?sample=styling-icon-fonts 
works fine in native JS but in Angular the objects are not displayed immediately. They are displayed only after performing Zoom operation (zoom in, zoom out).
In order to reproduce it, you can create an Angular project, copying the HTML code from the sample and see the issue.
They also claim that in JS API 4.17, it worked fine. Only 4.18 has this issue.

The question is why doesn't it work the same as in 4.17? i.e. Display the graphics immediately after adding them to the map, without the need in zoom operation.

Thanks,
Miri

**Comment by andygup** [comment]
Hi @MiriEshel, thanks for reporting. This is a known issue that we hope to address at 4.19 (April).

**Comment by jwerts** [comment]
I ran into the same thing and had put this together to demonstrate if it helps:  https://codepen.io/jwerts28/pen/JjbRVVw

**Comment by andygup** [comment]
Thanks for letting us know @jwerts.

**Comment by Cyberdada** [comment]
Hi, I'm running into the same problem :/
1) If removing and then adding featurelayer it does not redraw until zoom.
 
(if I do a timeout on first layer add I can see the layer.. something like this:

```
    void this.view.when().then(() => {
      setTimeout(() => {
        void this.view.goTo(this.realTimeLayer().graphics).then(() => {
       // do stuff...
        });
      }, 400);
    });
```
however after first draw if I try to click on something that I have drawn, it vanishes :/
( i have some code that changes color on a point marker, after vanishing the first time it appears, and everything works fine.)

April is pretty far away, is there someone that has a workaround for this until then?

**Comment by andygup** [comment]
Thanks for the additional info @Cyberdada. We are actively looking into this.

**Comment by andygup** [comment]
@MiriEshel @Cyberdada we believe we fixed this bug. Can you retest using `npm install @arcgis/core@next`?

**Comment by Cyberdada** [comment]
@andygup Now I get this error :
[esri.assets] The API assets location needs to be set using config.assetsPath.
:/ 
(assets have been configured in angular.json as described in documentation, this error did not come up in 4.18 ) 

This is the relevant part of the log:
Logger.js:5 [esri.assets] The API assets location needs to be set using config.assetsPath. More information: https://arcg.is/1OzLe50
_consoleWriter @ Logger.js:5
_log @ Logger.js:5
errorOnce @ Logger.js:5
a @ assets.js:5
get thumbnailUrl @ basemapDefinitions.js:5
u @ read.js:5
read @ JSONSupport.js:5
read @ Basemap.js:5
c @ JSONSupport.js:5
fromJSON @ JSONSupport.js:5
fromId @ Basemap.js:5
f @ basemapUtils.js:5
castBasemap @ Map.js:5
_cast @ Properties.js:5
set @ Properties.js:5
set @ subclass.js:5
t @ set.js:5
t @ set.js:5
set @ Accessor.js:5
postscript @ Accessor.js:5
p @ subclass.js:5

push../src/app/plugins/map/esri/services/esri-map-initialize.service.ts.EsriMapInitializeService.factory @ esri-map-initialize.service.ts:48

**Comment by Cyberdada** [comment]
It works :) I got a bit confused at first because a got an error about the assetsPath, that I did not get earlier, but then I checked out the documentation and added the assetsPath setting, now all is working great :)

**Comment by andygup** [comment]
Closing as fixed at 4.19.

**Comment by dougkpowers** [comment]
I'm still seeing this issue in 4.19.  Any confirmation that this has been fixed?

**Comment by andygup** [comment]
Hi @dougkpowers yes it's confirmed fixed. Do you have a codepen or github repo that reproduces the issue?

**Comment by dougkpowers** [comment]
@andygup I created a separate issue with full details and a codepen here: https://github.com/Esri/feedback-js-api-next/issues/122

I found the issue only occurs when Zone.js is imported.
